examples/integration/integration_test.go

```diff
@@ -1466,3 +1467,58 @@ func testResponseStrings(t *testing.T, port int) {
 	}
 
 }
+
+func TestRequestQueryParams(t *testing.T) {
+	testRequestQueryParamsKind(t, 8080, "GET")
+	testRequestQueryParamsKind(t, 8080, "POST")
+	testRequestQueryParamsKind(t, 8080, "PostForm")
+}
+
+func testRequestQueryParamsKind(t *testing.T, port int, kind string) {
+	var resp *http.Response
+	var err error
+	var apiURL, got, want string
+
+	switch kind {
+	case "GET":
+		apiURL = fmt.Sprintf("http://localhost:%d/v1/example/a_bit_of_everything/params/get/foo?double_value=%v&bool_value=%v", port, 1234.56, true)
+		resp, err = http.Get(apiURL)
+		want = `{"single_nested":{"name":"foo"},"double_value":1234.56,"bool_value":true}`
+	case "POST":
+		apiURL = fmt.Sprintf("http://localhost:%d/v1/example/a_bit_of_everything/params/post/hello-world?double_value=%v&bool_value=%v", port, 1234.56, true)
+		resp, err = http.Post(apiURL, "application/json", strings.NewReader(`{"name":"foo","amount":100}`))
+		want = `{"single_nested":{"name":"foo","amount":100},"double_value":1234.56,"bool_value":true,"string_value":"hello-world"}`
+	case "PostForm":
+		apiURL = fmt.Sprintf("http://localhost:%d/v1/example/a_bit_of_everything/params/get/foo?double_value=%v&bool_value=%v", port, 1234.56, true)
+		v := url.Values{}
+		v.Set("string_value", "hello-world")
+		v.Add("repeated_string_value", "demo1")
+		v.Add("repeated_string_value", "demo2")
+		resp, err = http.PostForm(apiURL, v)
+		want = `{"single_nested":{"name":"foo"},"double_value":1234.56,"bool_value":true,"string_value":"hello-world","repeated_string_value":["demo1","demo2"]}`
+	default:
+		t.Errorf("not support kind %q", kind)
+	}
+
+	if err != nil {
+		t.Errorf("http.Request(%q) failed with %v; want success", apiURL, err)
+		return
+	}
+	defer resp.Body.Close()
+
+	buf, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Errorf("ioutil.ReadAll(resp.Body) failed with %v; want success", err)
+		return
+	}
+	got = string(buf)
+
+	if got, want := resp.StatusCode, http.StatusOK; got != want {
+		t.Errorf("resp.StatusCode = %d; want %d", got, want)
+		t.Logf("%s", buf)
+	}
```

Comment:
Could you move this check above the Body check? The subscope `got` variable here is adding a bit of confusion.